huaweicloud/services/rds/resource_huaweicloud_rds_cross_region_backup_strategy.go

```diff
@@ -0,0 +1,328 @@
+		buildCreateBackupStrategyBodyParams(d))
+	if err != nil {
+		return diag.Errorf("error creating RDS cross region backup strategy: %s", err)
+	}
+
+	d.SetId(instanceID)
+
+	return resourceBackupStrategyRead(ctx, d, meta)
+}
+
+func buildCreateBackupStrategyBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"policy_para": map[string]interface{}{
+			"backup_type":            utils.ValueIngoreEmpty(d.Get("backup_type")),
+			"keep_days":              utils.ValueIngoreEmpty(d.Get("keep_days")),
+			"destination_region":     utils.ValueIngoreEmpty(d.Get("destination_region")),
+			"destination_project_id": utils.ValueIngoreEmpty(d.Get("destination_project_id")),
+		},
+	}
+	return bodyParams
+}
+
+func resourceBackupStrategyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getBackupStrategy: Query the RDS cross-region backup strategy
+	var (
+		getBackupStrategyHttpUrl = "v3/{project_id}/instances/{instance_id}/backups/offsite-policy"
+		getBackupStrategyProduct = "rds"
+	)
+	getBackupStrategyClient, err := cfg.NewServiceClient(getBackupStrategyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	instance, err := GetRdsInstanceByID(getBackupStrategyClient, d.Id())
+	if err != nil {
+		return diag.Errorf("error getting RDS instance: %s", err)
+	}
+	if instance.Id == "" {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	getBackupStrategyPath := getBackupStrategyClient.Endpoint + getBackupStrategyHttpUrl
+	getBackupStrategyPath = strings.ReplaceAll(getBackupStrategyPath, "{project_id}",
+		getBackupStrategyClient.ProjectID)
+	getBackupStrategyPath = strings.ReplaceAll(getBackupStrategyPath, "{instance_id}", d.Id())
+
+	getBackupStrategyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getBackupStrategyResp, err := getBackupStrategyClient.Request("GET", getBackupStrategyPath, &getBackupStrategyOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS cross region backup strategy")
+	}
+
+	getBackupStrategyRespBody, err := utils.FlattenResponse(getBackupStrategyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	policyPara := utils.PathSearch("policy_para", getBackupStrategyRespBody, nil)
+	if policyPara == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	backupStrategies := policyPara.([]interface{})
+	if len(backupStrategies) == 0 || utils.PathSearch("keep_days", backupStrategies[0], 0).(float64) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", d.Id()),
+		d.Set("backup_type", utils.PathSearch("backup_type", backupStrategies[0], nil)),
```

Comment:
it seems that we can delete this line as `backup_type` will be updated later.

Reply:
done